code/modules/almayer/weaponhits.dm

```diff
@@ -69,21 +88,91 @@
+					GLOB.ship_hits_tally["times_hit_railgun"] += 1
+					logtotalhits += 1
 				else
 					weaponhits_effects(WEAPON_RAILGUN, TRUE)
 
+		if(WEAPON_ODC)
+			var/datum/cause_data/orbitaldefensecannon_cause_data = create_cause_data("Orbital Defense Shell")
+			var/hitchance = HIT_CHANCE_CHEAT
+			if(point_defense == TRUE)
+				hitchance = HIT_CHANCE_STANDARD
+			if(salvo == TRUE)
+				var/confirmedhit
+				for(var/turf/picked_atom in location)
+					if(prob(hitchance))
+						cell_explosion(picked_atom, 850, 6, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, orbitaldefensecannon_cause_data)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_ODC_DAMAGE
+						shakeship(10, 10, TRUE, FALSE)
+						confirmedhit += 1
+						logtotalhits += 1
+				if(confirmedhit > 0)
+					weaponhits_effects(WEAPON_ODC)
+				if(confirmedhit < 1)
+					weaponhits_effects(WEAPON_ODC, TRUE)
+				GLOB.ship_hits_tally["times_hit_odc"] += confirmedhit
+
+			else if(salvo == FALSE)
+				if(prob(hitchance))
+					cell_explosion(location, 850, 6, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, orbitaldefensecannon_cause_data)
+					GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_ODC_DAMAGE
+					shakeship(10, 10, TRUE, FALSE)
+					weaponhits_effects(WEAPON_ODC)
+					GLOB.ship_hits_tally["times_hit_odc"] += 1
+					logtotalhits += 1
+				else
+					weaponhits_effects(WEAPON_ODC, TRUE)
+
+		if(WEAPON_AABOILER)
+			var/datum/cause_data/aaboiler_cause_data = create_cause_data("AA Boiler Shot")
+			var/hitchance = HIT_CHANCE_CHEAT
+			if(point_defense == TRUE)
+				hitchance = HIT_CHANCE_STANDARD
+			if(salvo == TRUE)
+				var/confirmedhit
+				for(var/turf/picked_atom in location)
+					if(prob(hitchance))
+						cell_explosion(picked_atom, 700, 9, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, aaboiler_cause_data)
+						GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_AABOILER_DAMAGE
+						shakeship(10, 10, FALSE, FALSE)
+						confirmedhit += 1
+						logtotalhits += 1
+				if(confirmedhit > 0)
+					weaponhits_effects(WEAPON_AABOILER)
+				if(confirmedhit < 1)
+					weaponhits_effects(WEAPON_AABOILER, TRUE)
+				GLOB.ship_hits_tally["times_hit_aaboiler"] += confirmedhit
+
+			else if(salvo == FALSE)
+				if(prob(hitchance))
+					cell_explosion(location, 700, 9, EXPLOSION_FALLOFF_SHAPE_EXPONENTIAL, null, aaboiler_cause_data)
+					GLOB.ship_health_vars["ship_hull_health"] = GLOB.ship_health_vars["ship_hull_health"] - WEAPON_AABOILER_DAMAGE
+					shakeship(10, 10, FALSE, FALSE)
+					weaponhits_effects(WEAPON_AABOILER)
+					GLOB.ship_hits_tally["times_hit_aaboiler"] += 1
+					logtotalhits += 1
+				else
+					weaponhits_effects(WEAPON_AABOILER, TRUE)
+	var/integerhitstally = GLOB.ship_hits_tally["times_hit_missile"] + GLOB.ship_hits_tally["times_hit_railgun"] + GLOB.ship_hits_tally["times_hit_odc"] + GLOB.ship_hits_tally["times_hit_aaboiler"]
```

Comment:
Above this line is where you could instead perform, rather than inside each weapon block:
`GLOB.ship_hits_tally[weaponused]++` if you do the first suggestion to redefine them.